### ISSUE TITLE
Threads test failure power+master

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#18555]
Test failure happens both on LLVM 3.81 and LLVM 3.9

```
Warning: threaded loop executed in order
LLVM ERROR: Cannot select: t40: i32,ch = AtomicLoadMin<Volatile LDST2[%66]> t9, t7, t13
  t7: i64,ch = load<LD8[%65](tbaa=!"jtbaa_immut")> t0, t4, undef:i64
    t4: i64 = add t2, Constant:i64<8>
      t2: i64,ch = CopyFromReg t0, Register:i64 %vreg36
        t1: i64 = Register %vreg36
      t3: i64 = Constant<8>
    t6: i64 = undef
  t13: i32 = AssertSext t11, ValueType:ch:i16
    t11: i32,ch = CopyFromReg t0, Register:i32 %vreg26
      t10: i32 = Register %vreg26
In function: julia_#11276###_threadsfor#503_74774
	From worker 5:	     * threads              Worker 5 terminated.
```

```
julia> versioninfo()
Julia Version 0.6.0-dev.676
Commit 628d0e4* (2016-09-16 16:11 UTC)
Platform Info:
  System: Linux (powerpc64le-unknown-linux-gnu)
  CPU: unknown
  WORD_SIZE: 64
  BLAS: libopenblas (NO_AFFINITY POWER8)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.0 (ORCJIT, pwr8)
```

**Comment by vchuravy** [comment]
https://llvm.org/bugs/show_bug.cgi?id=28818

**Comment by ViralBShah** [comment]
Also part of #17822, which we should leave open as it has a bounty on it.

**Comment by vchuravy** [comment]
fixed by #18557
